sgpt/app.py

```diff
@@ -77,15 +80,19 @@ def main(
         prompt = get_edited_prompt()
 
     if shell:
+        if system is not None:
+            raise typer.BadParameter("Cannot use --system with --shell.")
         # If default values, make response more accurate.
         if top_probability == 1 == temperature:
             temperature = 0.4
-        prompt = make_prompt.shell(prompt)
+        system, prompt = make_prompt.shell(prompt)
     elif code:
-        prompt = make_prompt.code(prompt)
+        if system is not None:
```

Comment:
```suggestion
        if not system:
```

Reply:
`if not role` triggers by default because `role` is by default `None`. I went with `if role:`